pkg/controller/nodes/dynamic/dynamic_workflow_test.go

```diff
@@ -329,4 +333,88 @@ func Test_dynamicNodeHandler_buildContextualDynamicWorkflow_withLaunchPlans(t *t
 		assert.Error(t, err)
 		assert.True(t, callsAdmin)
 	})
+	t.Run("dynamic wf cached", func(t *testing.T) {
```

Comment:
can we write a test for the failure case? Where the cache read fails, or the cache write fails

Reply:
done